course-mw/enrolment-actor/src/main/scala/org/sunbird/enrolments/ContentConsumptionActor.scala

```diff
@@ -193,6 +197,11 @@ class ContentConsumptionActor @Inject() extends BaseEnrolmentActor {
         val inputStatus = inputContent.getOrDefault(JsonKey.STATUS, 0.asInstanceOf[AnyRef]).asInstanceOf[Number].intValue()
         val updatedContent = new java.util.HashMap[String, AnyRef]()
         updatedContent.putAll(inputContent)
+        jsonFields.foreach(field =>
+          if (inputContent.containsKey(field))
+            updatedContent.put(field, mapper.writeValueAsString(inputContent.get(field)))
+
+        )
```

Comment:
Use `filter` to identify the Json keys from inputContent object and parse. After that, use putAll to set the map with parsed data.

Reply:
Resolved

Reply:
Added flatmap to build parse data.  Then used putAll to the map with parsed data.